### PR TITLE
Compute ticker for unlisted tokens

### LINKED ISSUE #33: Compute ticker for unlisted tokens
<a href="https://github.com/freeatnet"></img></a> **Issue by [freeatnet](https://github.com/freeatnet)**
_Wednesday Mar 07, 2018 at 16:47 GMT_
_Originally opened as https://github.com/forkdelta/backend-replacement/issues/72_

----

The set of tokens for which ticker is computed should be the intersection of the listed tokens and the tokens in active orders and trades.

### PR DESCRIPTION
**Pull request by JonathonDunford**
Closes #33 

This is so that our tickers database stores data on unlisted tokens that have active trading on the platform. This allows us to more accurately report volume in our APIs and provides user with a little more data. This also allows us to create some sort of front end interface of unlisted tokens if we decide to.

I created a function for getting all of the addresses for tokens that had orders placed or trades executed in the past 30 days. Then, combined that list with the list of tokens from the frontend configuration file.

I also simplified the `tokens` storage in app to only hold the addresses in an array.

Converted the token update time into a variable configurable in config.

### Todo
- [x] `app.py ` needs to be updated to append these extra tokens to the `tokens` list. This would be done after line 50 in `app.py`

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -46,8 +46,8 @@ def __str__(self):
         def updateTokens(self):
             try:
                 self._tokensUpdateTime = datetime.utcnow()
-                fd_config=requests.get(config.FRONTEND_CONFIG_FILE).json()
-                self._tokens = fd_config['tokens']
+                frontend_config=requests.get(config.FRONTEND_CONFIG_FILE).json()
```
Comment (Ebonsignori) [inline app/app.py]: Can we place the tokens list .json in a location that is separate from the backend and frontend, like in the tokenbase repo, or more easily the server? We can have a [simple python script](https://stackoverflow.com/questions/38594717/how-do-i-push-new-files-to-github) that updates the list and pushes it to the repo using ssh if we go the repo route. Updating a JSON located on the server with write access for the backend and read access for the frontend could be easier.

File: app/src/get_unlisted_tokens.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+async def get_unlisted_tokens():
+
+    created_after = datetime.utcnow() - timedelta(days=30)
+
+    where_args = (
+        ZERO_ADDR_BYTES,
+        created_after
+    )
+
+    async with App().db.acquire_connection() as conn:
```
Comment (Ebonsignori) [inline app/src/get_unlisted_tokens.py]: Could we, in addition to automatically adding new tokens, create a websocket endpoint that accepts the submission of a new token? For instance, on the frontend, we could create a simple interface where users can enter info about a new token (symbol, decimals, address, project, short description) and if the address isn't already in a token list, we add it to our "unverified tokens" list. We can then add an option for frontend users to select unverified tokens, but show them a warning and suggest that they verify the token address before trading them.

Comment (JonathonDunford) [reply app/src/get_unlisted_tokens.py]: @Ebonsignori Flesh out that idea and open in a separate issue.

File: app/app.py (hunk 1 above)
Comment (JonathonDunford) [reply app/app.py]: @Ebonsignori 

I would love to.

We should definitely have the new UI pull from `Tokens` as well as everything else.

Detail that in a new issue or something.

**Pull request merged**